packages/libircclient/patch-add-cloudabi

```diff
@@ -0,0 +1,406 @@
++ *                IRC server. Vast majority of IRC servers do not require passwords.
++ * \param nick    A nick, which libircclient will use to login to the IRC server.
++ *                Must not be NULL.
++ * \param username A username of the account, which is used to connect to the
++ *                IRC server. This is for information only, will be shown in
++ *                "user properties" dialogs and returned by /whois request.
++ *                May be NULL, in this case 'nobody' will be sent as username.
++ * \param realname A real name of the person, who connects to the IRC. Usually
++ *                people put some wide-available information here (URL, small
++ *                description or something else). This information also will 
++ *                be shown in "user properties" dialogs and returned by /whois 
++ *                request. May be NULL, in this case 'noname' will be sent as 
++ *                username.
++ *
++ * \return Return code 0 means success. Other value means error, the error
++ *  code may be obtained through irc_errno(). Any error, generated by the
++ *  IRC server, is available through irc_callbacks_t::event_numeric.
++ *
++ * This function adopts a file descriptor already connected to an IRC server.
++ * It performs an SSL handshake if necessary, then starts the handshake with
++ * the IRC server. You can use this function if you already have a file
++ * descriptor to an IRC server, and especially if your platform has no
++ * socket() or connect() function, e.g. CloudABI.
++ * 
++ * \sa irc_run
++ * \ingroup conndisc
++ */
++int irc_set_sockfd (irc_session_t * session,
++			int fd,
++			int ssl,
++			const char * server_password,
++			const char * nick,
++			const char * username,
++			const char * realname);
++
++/*!
+  * \fn void irc_disconnect (irc_session_t * session)
+  * \brief Disconnects a connection to IRC server.
+  *
+--- src/Makefile.in
++++ src/Makefile.in
+@@ -17,12 +17,14 @@ shared_mingw: libircclient.dll
+ 
+ install:
+ 	-mkdir -p $(DESTDIR)@prefix@/include
++	-mkdir -p $(DESTDIR)@prefix@/lib
+ 	-mkdir -p $(DESTDIR)@libdir@
+ 	cp ../include/libircclient.h $(DESTDIR)@prefix@/include/libircclient.h
+ 	cp ../include/libirc_errors.h $(DESTDIR)@prefix@/include/libirc_errors.h
+ 	cp ../include/libirc_events.h  $(DESTDIR)@prefix@/include/libirc_events.h 
+ 	cp ../include/libirc_options.h  $(DESTDIR)@prefix@/include/libirc_options.h 
+ 	cp ../include/libirc_rfcnumeric.h  $(DESTDIR)@prefix@/include/libirc_rfcnumeric.h 
++	cp libircclient.a $(DESTDIR)@prefix@/lib/libircclient.a
+ 	cp libircclient.so $(DESTDIR)@libdir@/libircclient.so.$(APIVERSION)
+ 	ln -fs libircclient.so.$(APIVERSION) $(DESTDIR)@libdir@/libircclient.so
+ 
+--- src/colors.c
++++ src/colors.c
+@@ -50,7 +50,7 @@ static inline void libirc_colorparser_addorcat (char ** destline, unsigned int *
+ 
+ 	if ( *destline )
+ 	{
+-		strcpy (*destline, str);
++		strlcpy (*destline, str, *destlen);
```

Comment:
To keep this package maintainable, I still suggest these changes are reverted and `unsafe_string_sources` is used instead.